model_struct.go

```diff
@@ -289,11 +289,24 @@ func (scope *Scope) GetModelStruct() *ModelStruct {
 									}
 
 									for _, name := range associationForeignKeys {
+
+										// In order to allow self-referencing many2many tables, the name
+										// may be followed by "=" to allow renaming the column
+										parts := strings.Split(name, "=")
+										name = parts[0]
+
 										if field, ok := toScope.FieldByName(name); ok {
 											// association foreign keys (db names)
 											relationship.AssociationForeignFieldNames = append(relationship.AssociationForeignFieldNames, field.DBName)
+
+											// If a new name was provided for the field, use it
+											name = field.DBName
+											if len(parts) > 1 {
+												name = parts[1]
+											}
+
 											// join table foreign keys for association
-											joinTableDBName := ToDBName(elemType.Name()) + "_" + field.DBName
+											joinTableDBName := ToDBName(elemType.Name()) + "_" + name
```

Comment:
Why not just `name`?

Reply:
This particular line will run even if an override is not specified. If I were to only assign `name` to `joinTableDBName`, then existing code would behave differently.

Reply:
Yes, but that can be changed.  If I set `ID=friend_id` then the column name should be `friend_id` not `user_friend_id`.

Reply:
I'm using your branch and made this change: https://github.com/jchavannes/gorm/commit/26342e

Also, thanks for creating this branch.  Self-referencing objects definitely needed to be supported.

Reply:
Oh, I see what you mean. Yes, I'd be fine with doing it that way and I can easily update this pull request if that is the preferred behavior.